src/main/java/com/pinterest/secor/uploader/GsUploadManager.java

```diff
@@ -0,0 +1,108 @@
+package com.pinterest.secor.uploader;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.FileContent;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.storage.Storage;
+import com.google.api.services.storage.StorageScopes;
+import com.google.api.services.storage.model.StorageObject;
+import com.pinterest.secor.common.LogFilePath;
+import com.pinterest.secor.common.SecorConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Collections;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+/**
+ * Manages uploads to Google Cloud Storage using the Storage class from the Google API SDK.
+ *
+ * It will use Service Account credential (json file) that can be generated from the Google Developers Console.
+ * By default it will look up for the default credential in the environment variable GOOGLE_APPLICATION_CREDENTIALS
+ * or it can be configured with secor.gs.credentials.path.
+ *
+ * Application credentials documentation
+ * https://developers.google.com/identity/protocols/application-default-credentials
+ *
+ * @author Jerome Gagnon (jerome.gagnon.1@gmail.com)
+ */
+public class GsUploadManager extends UploadManager {
+    private static final Logger LOG = LoggerFactory.getLogger(GsUploadManager.class);
+
+    private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+    private static final ExecutorService executor = Executors.newFixedThreadPool(256);
+
+    /**
+     * Global instance of the Storage. The best practice is to make it a single
+     * globally shared instance across your application.
+     */
+    private static Storage mStorageService;
+
+    private Storage mClient;
+
+    public GsUploadManager(SecorConfig config) throws Exception {
+        super(config);
+
+        String credentialPath = System.getenv("GOOGLE_APPLICATION_CREDENTIALS");
```

Comment:
The priorities should be reversed: get the creds from the properties first and fall back to the env.